lib/pbench/server/cache_manager.py

```diff
@@ -1029,20 +1029,32 @@ def subprocess_run(
                 # tar failures can include a message for each file in the
                 # archive, and can be quite large. Break stderr into lines, and
                 # gather those lines only up to our configured size limit.
+                #
+                # The exception we raise and ultimately log will start by
+                # identifying the tarball, followed by the command and the
+                # return code. The appended stderr may provide additional
+                # useful context, but in general beyond the first line it'll
+                # be fairly meaningless.
                 size = 0
                 message = []
                 lines = process.stderr.split("\n")
+                prefix = ""
                 for line in lines:
                     # Skip empty lines
                     if not line:
                         continue
                     size += len(line)
                     if size > MAX_ERROR:
+                        prefix = "[TRUNC]"
                         break
                     message.append(line)
 
-                # If even the first line was too big, truncate it
-                msg = "\n".join(message) if message else lines[0][:MAX_ERROR]
+                # Prefix with [TRUNC] if we're not including the entire message
+                # and truncate the first line if it was too long by itself.
+                if message:
+                    msg = prefix + ("\n".join(message))
+                else:
+                    msg = "[TRUNC]" + lines[0][:MAX_ERROR]
                 raise exception(
                     ctx,
                     f"{cmd[0]} exited with status {process.returncode}: {msg!r}",
```

Comment:
What we actually have is a long sequence of lines like `tar: fio__2020.11.06T07.23.47: Cannot mkdir: No space left on device` for each file in the tarball. These are kinda silly. I'd be very tempted to drop `stderr` entirely, but I would like to see at least one instance of the "No space left on device". I figured I might as well accumulate all the news that fits; but I think if I touch this again, I might just keep the first line from `stderr` which will tell us `No space left on device` and really gives us all we need to know beyond the exception's "unpack \<tarball>" header.

Reply:
> I figured I might as well accumulate all the news that fits; but I think if I touch this again, I might just keep the first line from stderr

I concur on the "all that fits" approach, which is why I think we should (at this level) be ignoring line breaks.

If you think including more than the first line from `stderr` is too much, I would reduce the `MAX_ERROR` rather than omitting additional lines (if they would otherwise fit).